smart-cards/src/main/java/gov/va/api/health/smartcards/patient/PatientController.java

```diff
@@ -85,11 +82,22 @@ void consumeBundle(B bundle, List<MixedEntry> target, Function<E, MixedEntry> mi
 
   private static List<CredentialType> credentialTypes(Parameters parameters) {
     checkRequestState(parameters.parameter() != null, "parameters are required");
-    return parameters.parameter().stream()
-        .filter(p -> "credentialType".equals(p.name()))
-        .map(Parameters.Parameter::valueUri)
-        .map(CredentialType::fromUri)
-        .collect(toList());
+    List<CredentialType> types =
+        parameters.parameter().stream()
+            .filter(p -> "credentialType".equals(p.name()))
+            .map(Parameters.Parameter::valueUri)
+            .map(CredentialType::fromUri)
+            .collect(toList());
+    validateCredentialTypes(types);
```

Comment:
do you want to validate before or after adding the default credential types?

Reply:
Before, since we're validating the client's request. In particular the empty/null check.